### PR TITLE
Refactor shared watch face code into base class

### LINKED ISSUE #431: Consolidate common watch face code
I'm thinking about refactoring the common parts of `WatchFaceDigital` and `WatchFaceAnalog` into a common base class. This should help reduce overall code size a little by eliminating duplicated code for similar functionality and make it easier for people to create new watch faces since they can start with some core features and not need to re-implement them themselves.

Everything related to drawing the watch face would be left up to the derived classes since that's the main thing that distinguishes them. The state management for things like date/time, battery charge level, BLE status, notifications, etc. would be provided by the base class.

Just looking for feedback on whether or not this seems worthwhile.

### PR DESCRIPTION
**Pull request by jonvmey**
Opening this up now to get some review feedback. I haven't tested it yet since I don't have access to my dev kit at the moment but I plan on putting it through its paces in the next week or so. Overall reduction in binary size was ~1.1kB, not huge but every bit helps and it will result in more overall savings as more watch faces are added.

Depending on if/when this PR and #334 get merged I'd also be happy to either help with converting the PineTimeStyle code to use the new interface or do the conversion myself.

Closes #431

### CODE REVIEW

**Comment by kieranc** [comment]
I'm happy to try make PineTimeStyle work with this but I'll probably wait until 334 is merged before I start working on it.

**Comment by Avamander** [comment]
Yesterday there was a discussion about PIC and shared libraries on the PineTime, based on this blog post: https://mcuoneclipse.com/2021/06/05/position-independent-code-with-gcc-for-arm-cortex-m/#comment-310677

The reason I'm mentioning this is _maybe_ it would be possible to take that into account now? So that watchfaces and their code could in theory be shared library, InfiniTime executes some exposed functions to keep the face alive. Or do you, @jonvmey , think they're too tangential at the moment?

**Comment by jonvmey** [comment]
Supporting shared libraries for the possibility of dynamic watchface loading in the future is a good idea. I think that if/when we want to tackle that it would probably be better off done separately from this set of changes.

**Comment by jonvmey** [comment]
Rebased my changes on top of latest `develop` and updated the `PineTimeStyle` code to use the base watch face as well. Couldn't find any differences in how any of the watchfaces behave before and after my changes. Would appreciate if anyone else has the time to stress test it to make sure I haven't missed anything.

File: src/displayapp/DisplayApp.h [hunk 1]
```diff
@@ -27,7 +27,7 @@ namespace Pinetime {
     class Settings;
     class Battery;
     class Ble;
-    class DateTime;
+    class DControllerateTime;
```
Comment (Riksu9000) [inline src/displayapp/DisplayApp.h:30]: Typo

Comment (jonvmey) [reply src/displayapp/DisplayApp.h]: Thanks, fixed.

**Comment by jonvmey** [comment]
Yeah, I'm happy to update this PR and fix the conflicts if it's going to get reviewed/merged.

**Comment by Quantum-cross** [comment]
if yall merge it using lvgl7, I'll fix it up and pull it into my 8 branch when that's done.

**Comment by lman0** [comment]
> Yeah, I'm happy to update this PR and fix the conflicts if it's going to get reviewed/merged

if you update this pr , you might have some chance that someone review it (even more if @Avamander allow action on your update)
maybe , once you have updated , @kieranc  or @geekbozu  might review/approve it making it more possible to be accepted by @JF002 .

in order to make it possible , you need to check every day , after ahaving  fix the pr , on github and discord if someone is willing to review your code and maybe show to @JF002  that you are 'alive" , otherwise he might forgot that you have finished the fix .

and for at least one month , you need to fix your pull each time @JF002 commit because it very unlikely that he merge instantly your pull once it have been fixed , revieved and approved since there is a bunch of @Riksu9000  pull waiting to be merged that he may merge before yours.

**Comment by Riksu9000** [comment]
I'm closing this due to being inactive. Please open a new PR if you want to continue working on this.

**Pull request closed without merging**